types/ember__object/evented.d.ts

```diff
@@ -10,9 +10,9 @@ interface Evented {
     on<Target>(
         name: string,
         target: Target,
-        method: (this: Target, ...args: any[]) => void
+        method: ((this: Target, ...args: any[]) => void) | string
     ): this;
-    on(name: string, method: (...args: any[]) => void): this;
+    on(name: string, method: ((...args: any[]) => void) | string): this;
```

Comment:
**Suuuuper nit:** for readability, can we flip the order of these? I always have a devil of a time parsing parenthesized function expressions like this. So `method: string | (this: Target, ...args: any[]) => void`, etc.